src/core/inline-idl-parser.js

```diff
@@ -9,75 +9,180 @@
  *  { base: "Foo.bar", method: "baz()", args: [] }
  * Foo.baz(arg1, arg2) ->
  *  { base: "Foo", method: "baz(arg1, arg2)", args: ["arg1", "arg2"] }
- * Dictionary["member"] ->
- *  { base: "Dictionary", member: "member" }
  */
 
-const methodRegex = /\((.*)\)$/;
-const idlSplitRegex = /\b\.\b|\.(?=\[\[)/;
-const dictionaryRegex = /(\w+)\["(\w+)"\]/;
-
 function parseInlineIDL(str) {
-  const result = Object.create(null);
-  const splitted = str.split(idlSplitRegex);
-  if (methodRegex.test(splitted[splitted.length - 1])) {
-    result.method = splitted.pop();
-    result.args = result.method.match(methodRegex)[1].split(/,\s*/);
-  }
-  if (splitted.length > 1 && !result.method) {
-    result.attribute = splitted.pop();
+  const methodRegex = /(\w+)\((.*)\)$/;
+  const slotRegex = /$\[\[(\w+)\]\]$/;
+  const attributeRegex = /^(\w+)$/;
+  const enumRegex = /^(\w+)\["([\w ]+)"\]$/;
+  const tokens = str.split(".");
```

Comment:
Lets split at `/\b\.\b/` to add support for variadic methods.

``` js
"foo.bar.baz(foo, ...bar)".split(".")
// ["foo", "bar", "baz(foo, ", "", "", "bar)"]
"foo.bar.baz(foo, ...bar)".split(/\b\.\b/)
// ["foo", "bar", "baz(foo, ...bar)"]
```

Reply:
Unfortunately, that one doesn't work well for:
"foo.bar.[[test]].baz(foo, ...bar)".split(/\b\.\b/)

We might need to split on method, then split the rest? Or so some kind of fancy look ahead, the split behind.

Reply:
```js
const splitRegex = /(?<=\]\]|\b)\./
```
😎 🤓